File: BaekJoon/투포인터/2470_두 용액.kt
fun main() {
        val sc = Scanner(System.`in`)
        val n = sc.nextInt()
        val array = IntArray(n) { sc.nextInt() }

        var left = 0
        var right = n - 1
        var standard = 2000000000

        var answer = IntArray(2) { 0 }

        array.sort()

        while (left < right) {
            var sum = array[left] + array[right]
            if (Math.abs(sum) < standard) {
                standard = Math.abs(sum)
                answer[0] = array[left]
                answer[1] = array[right]

                if (sum == 0) break
            }

            if (sum < 0) left++
            else right--
        }

        print("${answer[0]} ${answer[1]}")
    }

File: BaekJoon/투포인터/2470_두 용액.swift
let n = Int(readLine()!)!
var numbers = readLine()!.split(separator: " ").map { Int(String($0))! }

numbers.sort()

var start = 0
var end = numbers.count - 1
var goal = Int.max
var answer = [Int]()

while start < end {
    let sum = numbers[start] + numbers[end]
    
    if sum == 0 {
        answer = [numbers[start], numbers[end]]
        break
    }
    
    if abs(sum) < goal {
        goal = abs(sum)
        answer = [numbers[start], numbers[end]]
    }
    
    if sum < 0 { start += 1 }
    else { end -= 1 }
}


print(answer[0], answer[1])